### ISSUE TITLE
No registered users in connected users panel

### ISSUE THREAD

**Issue by pojlFDlxCOvZ4Kg8y1l4** [calzoneman/sync#969]
Please fill out the templates below to the best of your ability, based on whether your problem is with using the website or with running your own server.

## Website Problem ##
**Please confirm whether you've tried the following debugging steps:**

  - [X] Clearing cache and refreshing the page (On Firefox, press Ctrl+F5.  On Chrome, press F12, then right click the refresh button and click "Empty Cache and Hard Reload")
  - [X] Disabling all browser extensions
  - [X] Using a clean channel with no customizations
  
### Description of the Problem ###

  - What triggers the problem?
  Updating Cytube to the current version (2023-04-09, CyTube v3.85.1) from an older build (probably june 2022 or older)
  - What happens?
Registered users do not show up in the left panel of channel chat. Cytube expects you to register a temporary nime while logged in. 
Simply using your account name is refused, account already is in use.
  - What do you expect to happen instead?
  registered users in channel show up in the left pane, like before updating cytube. 

### System Information ###

  - **Operating System (Windows / Mac / Linux / Android / iOS):**
  Client: Windows 11
  Server: Ubuntu 20.04.6 LTS
  NPM 9.6.4
  - **Web Browser (Firefox / Chrome / Other):** 
  Current Chrome (Version 112.0.5615.49 (Official Build) (64-bit)) with no add-ons. 
  Current Firefox (111.0.1 (64-bit)) with add-ons.
  - **Error Messages Displayed:**
  
![image](https://user-images.githubusercontent.com/42654642/230774183-47f0e1b1-6d9e-462d-bf7a-c53fe381e7bc.png)

  - **Screenshot of JavaScript Console:**
  
![image](https://user-images.githubusercontent.com/42654642/230774157-5885ce88-5987-40bb-94bd-00cd112ffe5b.png)

**Comment by calzoneman** [comment]
From your first screenshot, it seems like the userlist is loading correctly, it's just that your browser is not being authenticated as your logged in account -- the behavior you described matches with how the page would appear to a logged-out user.

1. Can you confirm that you are able to use other logged-in functionality, and it's just the channel that isn't authenticating you?  For example, do you see the Account dropdown on the navbar, and are you able to access those pages normally?  If you log out and back in (or log in from a private browser window), then does it work?
2. Can you confirm that when you upgraded your server, you followed any relevant items in the [NEWS.md](https://github.com/calzoneman/sync/blob/3.0/NEWS.md) and ran `npm install` to rebuild the application and its dependencies?
3. Can you try clearing your browser cache and if relevant any website caches as well (e.g. if you are using Cloudflare, clear its cache)?  In the second screenshot, there is an error printed about trying to access a variable that doesn't exist, which may not directly be related to your problem, but it does indicate that you have mismatched versions of the client scripts -- `videojs.options.flash` is no longer referenced anywhere in `player.js`

**Comment by pojlFDlxCOvZ4Kg8y1l4** [comment]
1. I am able to use logged in functionality: 
![image](https://user-images.githubusercontent.com/42654642/230787365-2c2ae9d1-a9ee-41c2-944c-cb156228d82e.png)
But after logging out and back in, restarting browser, cache wipe etc. it still does not show my user in the left pane:
![image](https://user-images.githubusercontent.com/42654642/230787406-db5e2e0d-ec2a-4c8e-bed4-c07736deb0d9.png)

2. history shows following commands were used:
  285  git pull
  286  npm install
  287  npm run postinstall
But I did not see NEWS.md. I will look through that and will report back later today.
3. It may or may not be that Cloudflare tunnel is fiddling with the scripts. Development mode makes the error go away, it seems. I will come back and confirm this later today.

**Comment by calzoneman** [comment]
>It may or may not be that Cloudflare tunnel is fiddling with the scripts. Development mode makes the error go away, it seems. I will come back and confirm this later today.

If I recall, development mode disables all caching.  So you may just need to go to Caching > Configuration and purge the cache.

**Comment by pojlFDlxCOvZ4Kg8y1l4** [comment]
> If I recall, development mode disables all caching. So you may just need to go to Caching > Configuration and purge the cache.

Cache wipes do not work. 
I've dug around and tested some stuff, it seems Cloudflare's Rocket Loader does some weird stuff. 
Cloudflare's Developer mode enabled bypasses cache and with that Rocket Loader completely.
So I've fiddled with Rocket Loader.
With Rocket Loader enabled, it wants to use the deprecated `videojs.options.flash`.
With Rocket Loader disabled, it shows the connected users in the left pane.  
I've observed similar odd behaviour with Nextcloud, which also has Rocket Loader disabled.  

Globally rocket loader is still enabled for the domain, so other sites can use it.
I've created a page rule for Cytube, with following settings:
URL:  subdomain.domain.tld/*
Pick setting: Rocket loader
Toggle set to disabled, (x)
![image](https://user-images.githubusercontent.com/42654642/230794656-393c3bad-ea77-4aae-9b7b-f23cbbc86f66.png)

I'll try it out in the next few days, if this solves the problem then I think it's safe to say it's just Cloudflares Rocket Loader doing it's oddities once again.
